### PR TITLE
Update devfile/library to v1.0.0

### PR DESCRIPTION
**Pull request by kadel**
**What type of PR is this?**

/kind cleanup
/kind code-refactoring

**What does this PR do / why we need it**:

**Which issue(s) this PR fixes**:

Fixes: partily https://github.com/openshift/odo/issues/4589

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] Update changelog

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
everything should work as before, no user-facing changes

### CODE REVIEW

**Comment by kadel** [comment]
/retest

```
INFO[2021-06-02T17:26:53Z] [odo]  ✗  Failed to pull stack nodejs from registry.devfile.io/devfile-catalog/nodejs:latest with allowed media types [application/vnd.devfileio.devfile.layer.v1 image/png image/svg+xml application/vnd.devfileio.vsx.layer.v1.tar application/x-tar]: failed to copy: httpReaderSeeker: failed open: could not fetch content descriptor sha256:b72d04e51f3df4a1411b9d387a8cd80e4d9e5f2f410b7c8849c9e4a1e5ec148f (application/vnd.devfileio.devfile.layer.v1) from remote: not found 
```
Not sure where this came from :-( Maybe some temporary problem in devfile registry?
I run tests against local crc and it was ok.

File: pkg/odo/genericclioptions/context.go
Comment (feloy) [inline pkg/odo/genericclioptions/context.go:87]: Do you want to display it even if there are no warnings?

Comment (mik-dass) [inline pkg/odo/genericclioptions/context.go:87]: We can use the `log.Warningf()` here

**Comment by mik-dass** [comment]
There are more usage of `"github.com/devfile/library/pkg/devfile"` in 
https://github.com/openshift/odo/blob/6d551587635f2a70550854fcc85043ded5665faa/pkg/odo/cli/component/create.go#L16
https://github.com/openshift/odo/blob/6d551587635f2a70550854fcc85043ded5665faa/pkg/odo/cli/catalog/describe/component.go#L23

File: pkg/odo/genericclioptions/context.go
Comment (kadel) [reply pkg/odo/genericclioptions/context.go:87]: oh, good catch. That was debugging that I forgot to delete :innocent:

**Comment by kadel** [comment]
> There are more usage of `"github.com/devfile/library/pkg/devfile"` in
> https://github.com/openshift/odo/blob/6d551587635f2a70550854fcc85043ded5665faa/pkg/odo/cli/component/create.go#L16
> 
> 
> https://github.com/openshift/odo/blob/6d551587635f2a70550854fcc85043ded5665faa/pkg/odo/cli/catalog/describe/component.go#L23

I've updated it to use the new functions as well.

**Comment by maysunfaisal** [comment]
@kadel we should use the devfile/library releases https://github.com/devfile/library/releases instead of the master version. We are trying to release frequently so that it is easier for odo as odo is its biggest user now :)

If you wait till the end of the week or early next week, we plan on doing one more release because we want to release with devfile/api mod 2.1.0 as well(which is in master right now)

Btw I tried using the release version instead of master version in my PR as you may noticed in https://github.com/openshift/odo/pull/4730

**Comment by kadel** [comment]
> @kadel we should use the devfile/library releases https://github.com/devfile/library/releases instead of the master version. We are trying to release frequently so that it is easier for odo as odo is its biggest user now :)

@maysunfaisal 
I would like to use release tags as well, but the main reason for upgrading was support for devfile 2.1.0 and latest v1.0.0-alpha.3 still includes devfile/api/ v2.0.0 :-/

**Comment by kadel** [comment]
/retest
cluster setup issue
```
INFO[2021-06-03T15:28:57Z] level=error msg=Error: [ERR]: Error building changeset: InvalidChangeBatch: [Tried to create resource record set [name='api-int.ci-op-83b0n7i7-89922.origin-ci-int-aws.dev.rhcloud.com.', type='A'] but it already exists] 
```

**Comment by prietyc123** [comment]
re-provisioned a new minikube VM.

/test psi-kubernetes-integration-e2e

**Comment by yangcao77** [comment]
@kadel FYI, devfile/library just pushed a new release `v1.0.0` which support devfile/api v2.1.0. This should unblock you.

**Comment by mik-dass** [comment]
```
INFO[2021-06-07T11:48:34Z] {"component":"entrypoint","file":"prow/entrypoint/run.go:255","func":"k8s.io/test-infra/prow/entrypoint.gracefullyTerminate","level":"error","msg":"Process did not exit before 15s grace period","severity":"error","time":"2021-06-07T11:48:31Z"} 
INFO[2021-06-07T11:48:34Z] {"component":"entrypoint","error":"process timed out","file":"prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2021-06-07T11:48:31Z"} 
INFO[2021-06-07T11:48:34Z] error: failed to execute wrapped command: exit status 127 
INFO[2021-06-07T11:48:34Z]                                              
INFO[2021-06-07T11:48:34Z] Step unit-test-windows-unit-test-windows-steps failed after 2h0m30s. 
```
/retest

**Comment by maysunfaisal** [comment]
@kadel pls see my comment in the old PR https://github.com/openshift/odo/pull/4730#issuecomment-844336445 if this is an issue with the PSI tests. I found the tests were bugged and I'm not sure if they have been since fixed or your PR is facing the same issue..

**Comment by kadel** [comment]
> @kadel pls see my comment in the old PR [#4730 (comment)](https://github.com/openshift/odo/pull/4730#issuecomment-844336445) to check if this is an issue with the PSI tests. I found the tests were bugged and I'm not sure if they have since been fixed. or if your PR is facing the same issue..

Yeh, there was an issue with a test unrelated to this. It just got fixed in https://github.com/openshift/odo/pull/4784

**Comment by kadel** [comment]
/hold cancel
This  should be finaly ready

**Comment by kadel** [comment]
/test psi-unit-test-windows


<br class="Apple-interchange-newline">

**Review by mik-dass — APPROVED** [review]
/lgtm

**Pull request merged**
